pkg/controller/controller_broker.go

```diff
@@ -131,12 +131,39 @@ func shouldReconcileServiceBroker(broker *v1alpha1.ServiceBroker, now time.Time,
 			// The broker has a ready condition
 
 			if condition.Status == v1alpha1.ConditionTrue {
+
 				// The broker's ready condition has status true, meaning that
 				// at some point, we successfully listed the broker's catalog.
-				// We should reconcile the broker (relist the broker's
-				// catalog) if it has been longer than the configured relist
-				// interval since the broker's ready condition became true.
-				return now.After(condition.LastTransitionTime.Add(relistInterval))
+				if broker.Spec.RelistBehavior == v1alpha1.ServiceBrokerRelistBehaviorManual {
+					// If a broker is configured with RelistBehaviorManual, it should
+					// ignore the Duration and only relist based on spec changes
+
+					glog.V(10).Infof(
+						"Not processing ServiceBroker %v: RelistBehavior is set to Manual",
+						broker.Name,
+					)
+					return false
+				}
+
+				if broker.Spec.RelistDuration == nil {
```

Comment:
Is this check necessary given that we have already performed validation on the Broker Spec to ensure that the RelistDuration is not nil when the RelistBehavior is not Manual?

Reply:
I've been back and forth on this. Are folks generally comfortable with trusting and dereffing a pointer value given a validation?

Reply:
You should always program defensively, IMO.